src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBrokerFactory.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Android.App;
+using Android.Content;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.Internal.Broker;
+using Microsoft.Identity.Client.Internal.Requests;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.UI;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.Platforms.Android.Broker
+{
+    internal class AndroidBrokerFactory
+    {
+        private static BrokerType s_installedBroker;
```

Comment:
Are you sure this is ok? This implies that if user upgrades Authenticator but does not restart the app, it will use AccountManager mechanism. Not saying it's wrong, just wanted to double check that this is an acceptable experience.

Reply:
discussed offline.
This allows us to avoid having an extra handshake for each request and can be reset by restarting the app